src/background/runtime.js

```diff
@@ -44,10 +44,10 @@ class Runtime {
         this.permissions.notifications = true;
       }
       this.storage.local.preferences = message.payload.preferences;
-      if(message.payload.migrate) {
+      await this.storage.persist();
+      if (message.payload.migrate) {
         this.migration.onUpdate({previousVersion: message.payload.previousVersion});
```

Comment:
One last thing and we're good to merge. The `onUpdate` call should have an `await`, so that the migration is guaranteed to finish before the `initialize` in the options UI is called (it `await`s the `sendMessage` there).

Reply:
Oh, right. I forgot that. Thank you.